### ISSUE TITLE
Construct Siamese Network architecture.

### ISSUE THREAD

**Issue by dkoterwa** [SebChw/Actually-Robust-Training#36]
Because Siamese Networks share the same weights, we first have to prepare valid architecture in order to train it.

**Comment by SebChw** [comment]
Find some implementation maybe on different domain and experiment with it. Do some tutorials.

**Comment by ArturMakowski** [comment]
I found some Siamese network implementations specifically in the speaker identification domain but the majority of them are written in Tensorflow:
https://github.com/oscarknagg/voicemap #Tensorflow
https://github.com/gauthierdmn/speaker_identification_siamese #PyTorch 
https://github.com/rynczakd/development-of-methods-of-voice-biometrics-identification #Tensorflow

Also, I found Siamese network Pytorch implementation on the image MNIST dataset with triplet loss implementation
https://github.com/adambielski/siamese-triplet
https://colab.research.google.com/drive/1xT4O6iSIT5RKfhQB3kv2L-GrBoBUMnlH#scrollTo=16p2X9cx36W- #I took the most relevant code snippets from this repo and put it into google collab.

I found these articles helpful:
https://medium.com/analytics-vidhya/building-a-speaker-identification-system-from-scratch-with-deep-learning-f4c4aa558a56
https://medium.com/data-reply-it-datatech/speaker-verification-introduction-to-siamese-network-part-1-9cda9fabdf7
https://towardsdatascience.com/how-to-choose-your-loss-when-designing-a-siamese-neural-net-contrastive-triplet-or-quadruplet-ecba11944ec

**Comment by SebChw** [comment]
Take most promising notebook and try to use dataset from issue #35 to create a baseline

**Comment by Nyaa97** [comment]
May be helpful:
https://arxiv.org/pdf/2009.13480.pdf
https://arxiv.org/pdf/2112.04459.pdf

**Comment by Nyaa97** [comment]
https://huggingface.co/datasets/Nyaa97/art_sr_vc1_full
https://huggingface.co/datasets/Nyaa97/art_sr_vc1_mini1
https://huggingface.co/datasets/Nyaa97/art_sr_vc1_mini2
https://huggingface.co/datasets/Nyaa97/art_sr_vc1_test

**Comment by ArturMakowski** [comment]
https://colab.research.google.com/drive/1xT4O6iSIT5RKfhQB3kv2L-GrBoBUMnlH#scrollTo=yML7Q7YA2uA6
